File: dto/profile_dto.go
package dto

import (
	"time"

	"github.com/google/uuid"
)

type UpdateProfileRequest struct {
	DOB          string `json:"dob"` // YYYY-MM-DD
	Gender       string `json:"gender"`
	ProfilePhoto string `json:"profile_photo"`
	Languages    string `json:"languages"`
	Biography    string `json:"biography"`
}

type AddLicenseRequest struct {
	RegistrationNumber  string `json:"registration_number" binding:"required"`
	RegistrationCouncil string `json:"registration_council" binding:"required"`
	RegistrationYear    int    `json:"registration_year" binding:"required"`
	IssueDate           string `json:"issue_date"`
	ExpiryDate          string `json:"expiry_date"`
}

type LicenseResponse struct {
	LicenseID           uuid.UUID `json:"license_id"`
	RegistrationNumber  string    `json:"registration_number"`
	RegistrationCouncil string    `json:"registration_council"`
	RegistrationYear    int       `json:"registration_year"`
	LicenseStatus       string    `json:"license_status"`
	VerificationStatus  string    `json:"verification_status"`
	CreatedAt           time.Time `json:"created_at"`
}

type AddQualificationRequest struct {
	Degree         string `json:"degree" binding:"required"` // MBBS, MD, MS, DM, DNB
	Specialization string `json:"specialization"`
	College        string `json:"college" binding:"required"`
	University     string `json:"university" binding:"required"`
	YearCompleted  int    `json:"year_completed" binding:"required"`
}

type QualificationResponse struct {
	QualificationID uuid.UUID `json:"qualification_id"`
	Degree          string    `json:"degree"`
	Specialization  string    `json:"specialization"`
	College         string    `json:"college"`
	University      string    `json:"university"`
	YearCompleted   int       `json:"year_completed"`
	CreatedAt       time.Time `json:"created_at"`
}

type AddClinicRequest struct {
	ClinicName       string   `json:"clinic_name" binding:"required"`
	Address          string   `json:"address" binding:"required"`
	City             string   `json:"city" binding:"required"`
	State            string   `json:"state" binding:"required"`
	Pincode          string   `json:"pincode" binding:"required"`
	Latitude         *float64 `json:"latitude"`
	Longitude        *float64 `json:"longitude"`
	ConsultationMode string   `json:"consultation_mode" binding:"required"` // ONLINE, OFFLINE, BOTH
	ConsultationFee  float64  `json:"consultation_fee"`
}

type ClinicResponse struct {
	ClinicID         uuid.UUID `json:"clinic_id"`
	ClinicName       string    `json:"clinic_name"`
	Address          string    `json:"address"`
	City             string    `json:"city"`
	State            string    `json:"state"`
	Pincode          string    `json:"pincode"`
	ConsultationMode string    `json:"consultation_mode"`
	ConsultationFee  float64   `json:"consultation_fee"`
	CreatedAt        time.Time `json:"created_at"`
}

type DocumentUploadResponse struct {
	DocumentID       uuid.UUID `json:"document_id"`
	DoctorID         uuid.UUID `json:"doctor_id"`
	DocumentType     string    `json:"document_type"`
	FileURL          string    `json:"file_url"`
	OriginalFilename string    `json:"original_filename"`
	MIMEType         string    `json:"mime_type"`
	FileSize         int64     `json:"file_size"`
	FileHash         string    `json:"file_hash"`
	Version          int       `json:"version"`
	IsLatest         bool      `json:"is_latest"`
	OCRStatus        string    `json:"ocr_status"`
	UploadedAt       time.Time `json:"uploaded_at"`
}

type SubmitVerificationResponse struct {
	Message     string    `json:"message"`
	PublicID    uuid.UUID `json:"public_id"`
	Status      string    `json:"status"`
	SubmittedAt time.Time `json:"submitted_at"`
}

File: python_backend/dto/profile_dto.py
from pydantic import BaseModel, Field
from typing import Optional
from uuid import UUID
from datetime import datetime

class UpdateProfileRequest(BaseModel):
    dob: Optional[str] = Field(None, description="YYYY-MM-DD")
    gender: Optional[str] = None
    profile_photo: Optional[str] = None
    languages: Optional[str] = None
    biography: Optional[str] = None

class AddLicenseRequest(BaseModel):
    registration_number: str = Field(...)
    registration_council: str = Field(...)
    registration_year: int = Field(...)
    issue_date: Optional[str] = None
    expiry_date: Optional[str] = None

class LicenseResponse(BaseModel):
    license_id: UUID
    registration_number: str
    registration_council: str
    registration_year: int
    license_status: str
    verification_status: str
    created_at: datetime

class AddQualificationRequest(BaseModel):
    degree: str = Field(...)
    specialization: Optional[str] = None
    college: str = Field(...)
    university: str = Field(...)
    year_completed: int = Field(...)

class QualificationResponse(BaseModel):
    qualification_id: UUID
    degree: str
    specialization: str
    college: str
    university: str
    year_completed: int
    created_at: datetime

class AddClinicRequest(BaseModel):
    clinic_name: str = Field(...)
    address: str = Field(...)
    city: str = Field(...)
    state: str = Field(...)
    pincode: str = Field(...)
    latitude: Optional[float] = None
    longitude: Optional[float] = None
    consultation_mode: str = Field(...)
    consultation_fee: Optional[float] = None

class ClinicResponse(BaseModel):
    clinic_id: UUID
    clinic_name: str
    address: str
    city: str
    state: str
    pincode: str
    consultation_mode: str
    consultation_fee: float
    created_at: datetime

class DocumentUploadResponse(BaseModel):
    document_id: UUID
    doctor_id: UUID
    document_type: str
    file_url: str
    original_filename: str
    mime_type: str
    file_size: int
    file_hash: str
    version: int
    is_latest: bool
    ocr_status: str
    uploaded_at: datetime

class SubmitVerificationResponse(BaseModel):
    message: str
    public_id: UUID
    status: str
    submitted_at: datetime
